### ISSUE TITLE
rustfmt partially eats raw reference operators

### ISSUE THREAD

**Issue by tommit** [rust-lang/rustfmt#4066]
Input:
```rust
#![feature(raw_ref_op)]

fn main() {
    let x: i32 = 0;
    let _ = &raw const x as *const u32;

    let mut y: i32 = 0;
    let _ = &raw mut y as *mut u32;
}
```
Output:
```rust
#![feature(raw_ref_op)]

fn main() {
    let x: i32 = 0;
    let _ = &x as *const u32;

    let mut y: i32 = 0;
    let _ = &mut y as *mut u32;
}
```
The input compiles, while the output doesn't compile giving these errors:
```
error[E0606]: casting `&i32` as `*const u32` is invalid
error[E0606]: casting `&mut i32` as `*mut u32` is invalid
```
Using rustfmt 1.4.12-nightly (9f53665f 2020-02-10)

**Comment by calebcartwright** [comment]
Support for raw reference operator syntax was added in #4046, which has not yet been released.

@topecongiro - I can open a rustfmt v1.x PR with the commits that provide support for some of the newer syntax like this (plus trait const opt out and half open ranges) if you think we'll need to publish another 1.x release

**Comment by topecongiro** [comment]
Closing as this is fixed on the master branch and 1.4.13.
